api/v1beta2/cluster_types.go

```diff
@@ -536,22 +536,29 @@ type ClusterAvailabilityGate struct {
 	Polarity ConditionPolarity `json:"polarity,omitempty"`
 }
 
```

Comment:
@JoelSpeed Would be great to get your input on this API change. Thx!

Reply:
Moving to a ClassRef makes sense, some suggestions for improving the validation

Reply:
Thx!